### PR TITLE
Add GlobeControl

### PR DESCRIPTION
**Pull request by birkskyum**
I think we can use a simple GlobeControl

It just toggles between globe <-> mercator

https://github.com/user-attachments/assets/2e9a2fda-232f-4a61-9736-fa763c7ed852

I've drawn the globe svg, so there's no license issue

## Launch Checklist
 - [X] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!

### CODE REVIEW

File: src/css/svg/maplibregl-ctrl-globe.svg
Comment (HarelM) [inline src/css/svg/maplibregl-ctrl-globe.svg:1]: What's the copy right on this image? Did you create it yourself? If so, do you give MapLibre and anyone using maplibre the full rights to use it?

File: src/ui/control/globe_control.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ * A `GlobeControl` control contains a button for toggling the map projection between "mercator" and "globe".
+ *
+ * @group Markers and Controls
+ *
+ * @example
+ * ```ts
+ * let map = new Map()
+ *     .addControl(new GlobeControl());
+ * ```
+ */
```
Comment (HarelM) [inline src/ui/control/globe_control.ts:16]: Any examples that should use it? If so, update the examples and add a `@see` here.

File: src/css/svg/maplibregl-ctrl-globe.svg
Comment (birkskyum) [reply src/css/svg/maplibregl-ctrl-globe.svg:1]: Yes, as I wrote:
> I've drawn the globe svg, so there's no license issue

Don't know if it'll do, if I write here, like:

I'm hereby granting full copyright of the image to MapLibre Non-Profit Organization.

File: src/ui/control/globe_control.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+        this._updateGlobeIcon();
+        this._map.on('styledata', this._updateGlobeIcon);
+        return this._container;
+    }
+
+    /** {@inheritDoc IControl.onRemove} */
+    onRemove() {
+        DOM.remove(this._container);
+        this._map.off('styledata', this._updateGlobeIcon);
```
Comment (HarelM) [inline src/ui/control/globe_control.ts:39]: Should you unsubscribe from the `click` event as well here?

File: src/ui/control/globe_control.ts (hunk 1 above)
Comment (HarelM) [reply src/ui/control/globe_control.ts:16]: I see that you added it to `test/examples/globe-fill-extrusion.html`, great! Let's mention it here too.

File: src/ui/control/globe_control.test.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    beforeMapTest();
+    map = createMap();
+});
+
+afterEach(() => {
+    map.remove();
+});
+
+describe('GlobeControl', () => {
+    test('appears in top-right by default', () => {
```
Comment (HarelM) [inline src/ui/control/globe_control.test.ts:31]: Add a test to see that toggle clicking works as well.

File: src/css/svg/maplibregl-ctrl-globe.svg
Comment (HarelM) [reply src/css/svg/maplibregl-ctrl-globe.svg:1]: I believe it does, maybe there's a more elaborated standard way of writing this, IDK.

**Comment by HarelM** [comment]
I see that the coverage of the control is at 70%, can you get it higher to around 90% please?

**Comment by birkskyum** [comment]
it's at 100% coverage now

Screenshot 2024-11-01 at 21 22 37

**Review by HarelM — APPROVED** [review]
Great, thanks!!

**Pull request merged**